superscore/tests/test_window.py

```diff
@@ -80,3 +80,29 @@ def test_add_collection_refresh(qtbot: QtBot, test_client: Client):
     new_top_level_entries = new_entry_item.childCount()
 
     assert new_top_level_entries > orig_top_level_entries
+
+
+@setup_test_stack(sources=["db/filestore.json"], backend_type=FilestoreBackend)
+def test_take_snapshot(qtbot, test_client):
+    window = Window(client=test_client)
+    qtbot.addWidget(window)
+    collection = tuple(test_client.search(("uuid", "eq", UUID("a9f289d4-3421-4107-8e7f-2fe0daab77a5"))))[0]
+    snapshot = tuple(test_client.search(("uuid", "eq", UUID("ffd668d3-57d9-404e-8366-0778af7aee61"))))[0]
```

Comment:
```suggestion
    snapshot = tuple(test_client.search(("uuid", "eq", UUID("ffd668d3-57d9-404e-8366-0778af7aee61"))))
```

These are generators right?

Reply:
sorry my suggestion was nonsensical at first.  I've adjusted it to just use `next`

Reply:
I was just looking at that haha.  I think this is more pythonic.

Reply:
Unpacking the generator here is causing a problem in a subsequent call to Client.fill?...

Reply:
That's the implication, but I haven't yet isolated that happening

Reply:
So, the backend's entry cache goes from 10 entries down to 4 entries if I call `next` until reaching `StopIteration`.  Maybe the exception is preventing the cache from being restored?

Reply:
Ah I think I see the suspect code.  The backend's `.search` method yields from within a cache-modification decorator, so maybe when we don't exit that decorator immediately we end up throwing the cache into a weird intermediate state.

https://github.com/pcdshub/superscore/blob/dede6666a6f5761f8de5ef8913785827b19380c3/superscore/backends/filestore.py#L296-L315

I can't say I have a solution since I haven't reproduced the problem myself, but I'd bet it's in there somewhere

Reply:
The error persists without the cache.  However, calling `next` on both generators in the test a second time (resulting in `StopIteration`) fixes the error.  So I think the issue is ultimately nested `FilestoreBackend._load_and_store_context`s when the generator isn't exhausted.

Reply:
In that case I think I'll leave the code as-is.  `tuple(generator)` seems more intuitive than a hardcoded sequence of `next` calls.

Reply:
Sounds good, this is a separate bug to squash for sure.